File: uri/strings/complete_sentence/1551.cpp
#include <iostream>
#include <string>
#include <set>

using namespace std;

int main() {
    int n;
    cin >> n;
    cin.ignore();
    
    for (int i = 0; i < n; i++) {
        string line;
        getline(cin, line);
        
        set<char> letters;
        
        for (char c : line) {
            if (c >= 'a' && c <= 'z') {
                letters.insert(c);
            }
        }
        
        int count = letters.size();
        
        if (count == 26) {
            cout << "frase completa" << endl;
        } else if (count >= 13) {
            cout << "frase quase completa" << endl;
        } else {
            cout << "frase mal elaborada" << endl;
        }
    }
    
    return 0;
}

File: uri/strings/complete_sentence/1551.rs
use std::io::{self, BufRead};
use std::collections::HashSet;

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    
    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    for _ in 0..n {
        if let Some(Ok(line)) = lines.next() {
            let mut letters = HashSet::new();
            
            for c in line.chars() {
                if c >= 'a' && c <= 'z' {
                    letters.insert(c);
                }
            }
            
            let count = letters.len();
            
            if count == 26 {
                println!("frase completa");
            } else if count >= 13 {
                println!("frase quase completa");
            } else {
                println!("frase mal elaborada");
            }
        }
    }
}
